src/test/java/com/radixdlt/identifiers/RadixAddressTest.java

```diff
@@ -89,7 +90,7 @@ public void createAddressAndCheckUID() {
 
 	@Test(expected = CryptoException.class)
 	public void generateAddress() throws CryptoException {
-		new RadixAddress(magicByte(), new ECPublicKey(new byte[33]));
+		assertNotNull(new RadixAddress(magicByte(), new ECPublicKey(new byte[33])));
```

Comment:
Hmm this is a test where we are asserting that an error is thrown, thus `assertNotNull` call, which really is a call to assert that things are good is confusing for me. Also, how can it be not null, if an error is thrown?

not your code...: might be helpful to get a little bit of info of WHY we expect this to fail, bu splitting it up like this:

Also hmm the RadixAddress constructor will never be called right? Because the error will be thrown in the constructor of the `ECPublicKey` right? I really think we should move this test to `ECPublicKeyTest`

```java
byte[] invalidPublicKeyBytes = new byte[33]
ECPublicKey invalidPublicKey = new ECPublicKey(invalidPublicKeyBytes))
fail()
```

Reply:
We could do this, but constructing objects that are never used and declaring local variables that are never referenced are both compiler warnings, which this change was trying to remove.

Have changed all of these to use `assertThatThrownBy(...)` from AssertJ, which resolved these issues in a different way.